android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapViewManager.java

```diff
@@ -245,20 +246,27 @@ public void setUserLocationVerticalAlignment(RCTMGLMapView mapView, int userLoca
     public static final int METHOD_GET_ZOOM = 8;
     public static final int METHOD_GET_CENTER = 9;
 
+    /**
+     * make this overridable for subclasses to add their own methods
+     */
+    protected Builder<String, Integer> getMapBuilder(){
+        return MapBuilder.<String, Integer>builder()
+            .put("setCamera", METHOD_SET_CAMERA)
+            .put("queryRenderedFeaturesAtPoint", METHOD_QUERY_FEATURES_POINT)
+            .put("queryRenderedFeaturesInRect", METHOD_QUERY_FEATURES_RECT)
+            .put("getVisibleBounds", METHOD_VISIBLE_BOUNDS)
+            .put("getPointInView", METHOD_GET_POINT_IN_VIEW)
+            .put("getCoordinateFromView", METHOD_GET_COORDINATE_FROM_VIEW)
+            .put("takeSnap", METHOD_TAKE_SNAP)
+            .put("getZoom", METHOD_GET_ZOOM)
+            .put("getCenter", METHOD_GET_CENTER);
+
+    }
```

Comment:
This will allow us to add our own custom methods on the child class. The child can simply override this method and add his method declarations to this collection.